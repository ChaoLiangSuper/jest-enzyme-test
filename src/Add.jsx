import React from 'react';
import { add } from './functions';

class Add extends React.Component {
  render() {
    const { a, b } = this.props;
    return <div>add: {add(a, b)}</div>
  }
}

Add.defaultProps = {
  a: 0,
  b: 0
};

export default Add;
