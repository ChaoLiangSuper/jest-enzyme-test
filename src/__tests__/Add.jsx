import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Add from '../Add';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const props = {
  a: 1,
  b: 1
}

it('should render default component', () => {
  const component = renderer.create(<Add />);
  const snap = component.toJSON();
  expect(snap).toMatchSnapshot();
});

it('should render with provided props', () => {
  const component = renderer.create(<Add {...props} />);
  const snap = component.toJSON();
  expect(snap).toMatchSnapshot();
});

it('should render expected html', () => {
  const component = shallow(<Add {...props} />);
  expect(component.html()).toBe('<div>add: 2</div>');
});
