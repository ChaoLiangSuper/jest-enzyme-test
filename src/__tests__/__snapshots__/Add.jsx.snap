// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default component 1`] = `
<div>
  add: 
  0
</div>
`;

exports[`should render with provided props 1`] = `
<div>
  add: 
  2
</div>
`;
